utils/raft.go

```
@@ -57,181 +76,1347 @@ type VoteResponse struct {
+				rn.mu.Unlock()
+				votingComplete = true
+			}
+		}
+	}()
+
+	// Send RequestVote RPCs to other nodes
 	for _, peer := range rn.peers {
 		go func(peer string) {
-			conn, err := grpc.Dial("localhost:"+peer, grpc.WithInsecure())
+			// Try to connect to the peer
+			conn, err := grpc.Dial(peer, grpc.WithInsecure())
```

golangci-lint (1.64.8)

SA1019: grpc.Dial is deprecated: use NewClient instead.  Will be supported throughout 1.x. 

(staticcheck)